exercise.wwwapi/Endpoints/PostsEndpoint.cs

```diff
@@ -0,0 +1,67 @@
+using exercise.wwwapi.Models;
+using exercise.wwwapi.Repository;
+using exercise.wwwapi.DTO;
+using Microsoft.AspNetCore.Authorization;
+using System.Security.Claims;
+using exercise.wwwapi.Helpers;
+namespace exercise.wwwapi.Endpoints
+{
+    public static class PostsEndpoint
+    {
+        public static void ConfigureApi(this WebApplication app)
+        {
+            app.MapGet("/posts", GetAllPosts);
+            app.MapPost("/posts", CreatePost);
+            app.MapPut("/posts", UpdatePost);
+        }
+        //Automatically checks in the header of the request if a jwt token is included and if it is valid
+        //if so, we retrieve posts, else we get unauthorized response
+        [Authorize()]
+        private static async Task<IResult> GetAllPosts(IRepository repository)
+        {
+            return TypedResults.Ok(PostsResponseDTO.FromRepository(await repository.GetPosts()));
+        }
+
+        [Authorize()]
+        private static async Task<IResult> CreatePost(IRepository repository, PostsPostPayload payload, ClaimsPrincipal user)
+        {
+            //retrieves the user id from the token which is included from the header by the Authorize() method
+            string? userId = user.UserId();
+            if (userId == null)
+            {
+                return TypedResults.Unauthorized();
+            }
+            Posts post = await repository.CreatePost(payload.text, userId);
+            return TypedResults.Ok(post);
+        }
+
+        [Authorize()]
```

Comment:
If you are enforcing the admin role, then the decorator should be [Authorize(Role="Administrator")]